app/src/main/java/com/android/sample/ui/composables/cameraFeedback.kt

```diff
@@ -0,0 +1,131 @@
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.viewinterop.AndroidView
+import androidx.core.content.ContextCompat
+import androidx.lifecycle.compose.LocalLifecycleOwner
+import com.android.sample.R
+import com.android.sample.model.camera.CameraViewModel
+import com.google.accompanist.permissions.ExperimentalPermissionsApi
+import com.google.accompanist.permissions.PermissionState
+import com.google.accompanist.permissions.isGranted
+import com.google.accompanist.permissions.rememberPermissionState
+import java.io.File
+
+class CameraFeedBack {
+
+  companion object {
+    private var recording: Recording? = null
+
+    @androidx.annotation.OptIn(ExperimentalVideo::class)
+    @OptIn(ExperimentalPermissionsApi::class)
+    @Composable
+    public fun CameraScreen(cameraViewModel: CameraViewModel, modifier: Modifier = Modifier) {
+
+      val cameraPermissionState: PermissionState =
+          rememberPermissionState(android.Manifest.permission.CAMERA)
+
+      if (cameraPermissionState.status.isGranted) {
+        Box(modifier = modifier) { CameraBody(cameraViewModel) }
+      } else {
+        LaunchedEffect(Unit) { cameraPermissionState.launchPermissionRequest() }
+      }
+    }
+
+    @androidx.annotation.OptIn(ExperimentalVideo::class)
+    @Composable
+    fun CameraBody(cameraViewModel: CameraViewModel) {
+
+      val context = LocalContext.current
+      val lifecycleOwner = LocalLifecycleOwner.current
+
+      Scaffold(
+          modifier = Modifier.fillMaxSize(),
+          floatingActionButton = {
+            FloatingActionButton(
+                onClick = { cameraViewModel.switchCamera() }, modifier = Modifier.size(30.dp)) {
+                  Image(
+                      painter = painterResource(id = R.drawable.baseline_flip_camera_android_24),
+                      contentDescription = "Switch camera")
+                }
+          }) { pd: PaddingValues ->
+            AndroidView(
+                modifier = Modifier.padding(pd).fillMaxSize(),
+                factory = { context ->
+                  PreviewView(context)
+                      .apply {
+                        layoutParams =
+                            LinearLayout.LayoutParams(
+                                ViewGroup.LayoutParams.MATCH_PARENT,
+                                ViewGroup.LayoutParams.MATCH_PARENT)
+                        setBackgroundColor(android.graphics.Color.BLACK)
+                        scaleType = PreviewView.ScaleType.FIT_CENTER
+                      }
+                      .also { previewView ->
+                        previewView.controller = cameraViewModel.cameraController.value
+                        cameraViewModel.cameraController.value.bindToLifecycle(lifecycleOwner)
```

Comment:
Again maybe it would be good to release the resources and unbind within a DisposableEffect.

Reply:
Sorry my above message is quite vague, what I meant to say is to do something like this:
`cameraViewModel.cameraController.value.unbind()`
so that we can clear the lifecycleOwner. For this you could add a DisposableEffect.
[Here is the documentation for reference.](https://developer.android.com/reference/androidx/camera/view/LifecycleCameraController)